op-node/p2p/gossip.go

```diff
@@ -160,10 +165,21 @@ func NewGossipSub(p2pCtx context.Context, h host.Host, cfg *rollup.Config, gossi
 		pubsub.WithBlacklist(denyList),
 		pubsub.WithGossipSubParams(params),
 		pubsub.WithEventTracer(&gossipTracer{m: m}),
+		pubsub.WithPeerScoreInspect(BuildPeerScoreInspector(m), peerScoreInspectFrequency),
 	}
 	gossipOpts = append(gossipOpts, gossipConf.ConfigureGossip(&params)...)
 	return pubsub.NewGossipSub(p2pCtx, h, gossipOpts...)
-	// TODO: pubsub.WithPeerScoreInspect(inspect, InspectInterval) to update peerstore scores with gossip scores
+}
+
+// BuildPeerScoreInspector returns a function that is called periodically by the pubsub library to inspect the peer scores.
+// It is passed into the pubsub library as a [pubsub.ExtendedPeerScoreInspectFn] in the [pubsub.WithPeerScoreInspect] option.
```

Comment:
When you put a type in brackets, does it create a hyperlink in the godoc?

Reply:
great comment btw

Reply:
It should! See: https://tip.golang.org/doc/comment
<img width="1234" alt="Screenshot 2023-01-19 at 8 31 12 PM" src="https://user-images.githubusercontent.com/21288394/213598839-1dfebbd8-6275-4604-9491-1fdd0b7c86e2.png">